hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegionServer.java

```diff
@@ -789,8 +789,19 @@ public boolean registerService(com.google.protobuf.Service instance) {
     return true;
   }
 
-  private Configuration unsetClientZookeeperQuorum() {
+  private Configuration cleanupConfiguration() {
     Configuration conf = this.conf;
+    // We use ZKConnectionRegistry for all the internal communication, primarily for these reasons:
+    // - Decouples RS and master life cycles. For example, if all the masters are down, region
+    //   servers can abort at the same time, because the internal connection is master dependent and
+    //   fails.This is an operational nightmare. Using the ZK based registry means that the region
+    //   servers are now on the look out for new masters, if they are spun up.
+    // - Configuration management for region servers (cluster internal) is much simpler when adding
+    //   new masters etc.
+    // - We need to retain ZKConnectionRegistry for replication use anyway, so we just extend it for
+    //   other internal connections too.
+    conf.set(HConstants.CLIENT_CONNECTION_REGISTRY_IMPL_CONF_KEY,
+        "org.apache.hadoop.hbase.client.ZKConnectionRegistry");
```

Comment:
Ok, but these class names can be constant strings at least.

Reply:
Done.